common/common.cpp

```diff
@@ -139,6 +139,18 @@ bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
     return result;
 }
 
+constexpr std::uint64_t constexpr_hash(const char* input) {
+    std::uint64_t hash = 0xcbf29ce484222325;
+    const std::uint64_t prime = 0x100000001b3;
+
+    for (; *input; ++input) {
+        hash ^= static_cast<std::uint64_t>(*input);
+        hash *= prime;
+    }
+
+    return hash;
+}
```

Comment:
This is C++14 - we support just C++11 (https://stackoverflow.com/questions/45097171/the-body-of-constexpr-function-not-a-return-statement)

Probably, just remove the `else`s from the if-else chain - this is not performance critical code